db_test.go

```diff
@@ -532,10 +532,73 @@ func withRAMIdxDB(t *testing.T, fn func(t *testing.T, db *DB)) {
 }
 
 func withBPTSpareeIdxDB(t *testing.T, fn func(t *testing.T, db *DB)) {
-	tmpdir, _ := ioutil.TempDir("", "nutsdb")
+	tmpdir, _ := os.MkdirTemp("", "nutsdb")
 	opt := DefaultOptions
 	opt.Dir = tmpdir
 	opt.EntryIdxMode = HintKeyAndRAMIdxMode
 
 	withDBOption(t, opt, fn)
 }
+
+func Test_Three_EntryIdexMode_RestartDB(t *testing.T) {
+
+	tmpdir, _ := os.MkdirTemp("", "nutsdb")
+	opt := DefaultOptions
+	opt.Dir = tmpdir
+	opt.EntryIdxMode = HintKeyValAndRAMIdxMode
+	opt.SegmentSize = 8 * 1024
+
+	withDBOption(t, opt, func(t *testing.T, db *DB) {
```

Comment:
I can recall that @bigboss2063 refacted the code which runs test case with customized database. Could you help also refactor this part of code when you have time? For more context and code style you can refer this artcle.  Thanks！https://mp.weixin.qq.com/s/w8YaLU0b3bEzUXdrxezodQ

Reply:
And this is the related PR link!  https://github.com/nutsdb/nutsdb/pull/375

Reply:
ok